### PR TITLE
feature1

### PR DESCRIPTION
**Pull request by Jian-Min-Huang**
Add feature1

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.idea
```
Comment (Jian-Min-Huang-Test) [inline .gitignore]:
```suggestion
maybe add some comment here ?
like #Intellij IDEA
```

Comment (Jian-Min-Huang-Test) [inline .gitignore]:
```suggestion
# Intellij IDEA
.idea
```

File: src/org/yfr/App.java [hunk 2]
```diff
@@ -3,7 +3,13 @@
...
 
     public static void main(String[] args) {
-        System.out.println("Hi");
+        // setup slack app
+
+        // define handler
+
+        // listen and start
+
+            System.out.println("Hi");
```
Comment (Jian-Min-Huang-Test) [inline src/org/yfr/App.java]: why tab here ?

File: src/org/yfr/App.java [hunk 3]
```diff
@@ -3,7 +3,13 @@
 public class App {
 
     public static void main(String[] args) {
-        System.out.println("Hi");
+        // setup slack app
+
```
Comment (Jian-Min-Huang-Test) [inline src/org/yfr/App.java]: real case ?

File: src/org/yfr/App.java [hunk 4]
```diff
@@ -3,7 +3,13 @@
...
+        // define handler
+
```
Comment (Jian-Min-Huang-Test) [inline src/org/yfr/App.java]: real case ?

File: src/org/yfr/App.java [hunk 5]
```diff
@@ -3,7 +3,13 @@
...
+        // listen and start
+
```
Comment (Jian-Min-Huang-Test) [inline src/org/yfr/App.java]: real case ?

File: src/org/yfr/App.java (hunk 2 above)
Comment (Jian-Min-Huang) [reply src/org/yfr/App.java]: my foul, fix later

File: src/org/yfr/App.java (hunk 3 above)
Comment (Jian-Min-Huang) [reply src/org/yfr/App.java]: I'll add it

File: src/org/yfr/App.java (hunk 4 above)
Comment (Jian-Min-Huang) [reply src/org/yfr/App.java]: I'll add it

File: src/org/yfr/App.java (hunk 5 above)
Comment (Jian-Min-Huang) [reply src/org/yfr/App.java]: I'll add it

File: src/org/yfr/App.java [hunk 6]
```diff
@@ -4,12 +4,20 @@
 
     public static void main(String[] args) {
         // setup slack app
+        // App expects env variables (SLACK_BOT_TOKEN, SLACK_SIGNING_SECRET)
+        // App app = new App();
 
         // define handler
+//        app.command("/hello", (req, ctx) -> {
+//            return ctx.ack(":wave: Hello!");
+//        });
```
Comment (Jian-Min-Huang-Test) [inline src/org/yfr/App.java]:
```suggestion
        // app.command("/hello", (req, ctx) -> {
        //     return ctx.ack(":wave: Hello!");
        // });
```

File: src/org/yfr/App.java [hunk 7]
```diff
@@ -4,12 +4,20 @@
...
+
 
         // listen and start
+//        SlackAppServer server = new SlackAppServer(app);
+//        server.start(); // http://localhost:3000/slack/events
```
Comment (Jian-Min-Huang-Test) [inline src/org/yfr/App.java]:
```suggestion
        // SlackAppServer server = new SlackAppServer(app);
        // server.start(); // http://localhost:3000/slack/events
```

**Pull request merged**